internal/provider/resource_workspace.go

```diff
@@ -0,0 +1,330 @@
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+			},
+			"group_path": {
+				Type:                types.StringType,
+				MarkdownDescription: "Path of the parent group.",
+				Description:         "Path of the parent group.",
+				Required:            true,
+			},
+			"max_job_duration": {
+				Type:                types.Int64Type,
+				MarkdownDescription: "Maximum job duration in minutes.",
+				Description:         "Maximum job duration in minutes.",
+				Optional:            true,
+			},
+			"terraform_version": {
+				Type:                types.StringType,
+				MarkdownDescription: "Terraform version for this workspace.",
+				Description:         "Terraform version for this workspace.",
+				Optional:            true,
+			},
+			"prevent_destroy_plan": {
+				Type:                types.BoolType,
+				MarkdownDescription: "Whether a destroy plan would be prevented.",
+				Description:         "Whether a destroy plan would be prevented.",
+				Optional:            true,
+			},
+			"last_updated": {
+				Type:                types.StringType,
+				MarkdownDescription: "Timestamp when this workspace was most recently updated.",
+				Description:         "Timestamp when this workspace was most recently updated.",
+				Computed:            true,
+			},
+		},
+	}, nil
+}
+
+// Configure lets the provider implement the ResourceWithConfigure interface.
+func (t *workspaceResource) Configure(_ context.Context,
+	req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	t.client = req.ProviderData.(*tharsis.Client)
+}
+
+func (t *workspaceResource) Create(ctx context.Context,
+	req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	// Retrieve values from workspace.
+	var workspace WorkspaceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &workspace)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create the workspace.
+	created, err := t.client.Workspaces.CreateWorkspace(ctx,
+		&ttypes.CreateWorkspaceInput{
+			Name:               workspace.Name.ValueString(),
+			Description:        workspace.Description.ValueString(),
+			GroupPath:          workspace.GroupPath.ValueString(),
+			MaxJobDuration:     ptr.Int32(int32(workspace.MaxJobDuration.ValueInt64())),
+			TerraformVersion:   ptr.String(workspace.TerraformVersion.ValueString()),
+			PreventDestroyPlan: ptr.Bool(workspace.PreventDestroyPlan.ValueBool()),
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating workspace",
+			err.Error(),
+		)
+		return
+	}
+
+	// Map the response body to the schema and update the plan with the computed attribute values.
+	// Because the schema uses the Set type rather than the List type, make sure to set all fields.
+	t.copyWorkspace(*created, &workspace)
+
+	// Set the response state to the fully-populated plan.
+	resp.Diagnostics.Append(resp.State.Set(ctx, workspace)...)
+	if resp.Diagnostics.HasError() {
+		return
```

Comment:
This return statement is not needed, the same applies to any other function that has code similar to this.